test/pj-rehearse-integration/run.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# This test sets up a temporary testing repository
+# resembling `openshift/release` and tries to execute
+# pj-rehearse as if run over a PR for that repo
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+workdir="$( mktemp -d )"
+trap 'rm -rf "${workdir}"' EXIT
+
+test_root="$( dirname "${BASH_SOURCE[0]}")"
+master_data="${PWD}/${test_root}/master"
+candidate_data="${PWD}/${test_root}/candidate"
+fake_openshift_release="${workdir}/repo"
+expected_rehearsed_jobs="${test_root}/expected_jobs"
+
+echo "[INFO] Preparing fake input repository..."
+mkdir "${fake_openshift_release}"
+pushd "${fake_openshift_release}" >/dev/null
+git init --quiet
+cp -R "${master_data}"/* .
+git add ci-operator cluster
+git commit -m "Master version of openshift/release" --quiet
+base_sha="$(git rev-parse HEAD)"
+cp -R "${candidate_data}"/* .
+git add ci-operator cluster
+git commit -m "Candidate version of openshift/release" --quiet
+candidate_sha="$(git rev-parse HEAD)"
+popd >/dev/null
+
+echo "[INFO] Running pj-rehearse in dry-mode..."
+submitted_yamls="${workdir}/rehearsals.yaml"
+error_log="${workdir}/errors.log"
+export JOB_SPEC='{"type":"presubmit","job":"pull-ci-openshift-release-master-rehearse","buildid":"0","prowjobid":"uuid","refs":{"org":"openshift","repo":"release","base_ref":"master","base_sha":"'${base_sha}'","pulls":[{"number":1234,"author":"petr-muller","sha":"'${candidate_sha}'"}]}}'
```

Comment:
Oh boy

Reply:
Maybe I'll extract it to a function so it does not clutter the flow that much.